odoo/orm/domains.py

```diff
@@ -1285,22 +1285,30 @@ def _optimize_type_date(condition, _):
 
 
 def _value_to_datetime(value):
+    """Convert a value(s) to datetime.
+
+    :returns: A tuple containing the converted value and a boolean indicating
+              that all input values were dates.
+              These are handled differently during rewrites.
```

Comment:
Does this line mean the second return value technically and semantically might be changed in the recent future e.g. if we want to handle equality of date?

Reply:
To change the semantic and handle the quality, the `delta` variable needs to be updated baed on the `is_date` variable. Just like for inequalities.